### ISSUE TITLE
Too big of a link-time expression causes a segfault

### ISSUE THREAD

**Issue by mid-kid** [rednex/rgbds#334]
What it says on the tin.

Reproducing:
```asm
SECTION "a", ROM0
aa:

SECTION "b", ROM0
bb:

SECTION "c", ROM0
cc:

SECTION "d", ROM0
dd:

SECTION "e", ROM0
ee:

SECTION "f", ROM0
ff:

SECTION "g", ROM0
gg:

SECTION "h", ROM0
hh:

SECTION "i", ROM0
ii:

SECTION "j", ROM0
jj:

SECTION "k", ROM0
kk:

SECTION "l", ROM0
ll:

SECTION "m", ROM0
mm:

SECTION "n", ROM0
nn:

SECTION "o", ROM0
oo:

SECTION "p", ROM0
pp:

SECTION "z", ROM0
zz:

SECTION "end", ROM0

test: MACRO
    db ((BANK(\1) == BANK(aa)) * 1 + \
        (BANK(\1) == BANK(bb)) * 2 + \
        (BANK(\1) == BANK(cc)) * 3 + \
        (BANK(\1) == BANK(dd)) * 4 + \
        (BANK(\1) == BANK(ee)) * 5 + \
        (BANK(\1) == BANK(ff)) * 6 + \
        (BANK(\1) == BANK(gg)) * 7 + \
        (BANK(\1) == BANK(hh)) * 8 + \
        (BANK(\1) == BANK(ii)) * 9 + \
        (BANK(\1) == BANK(jj)) * 10 + \
        (BANK(\1) == BANK(kk)) * 11 + \
        (BANK(\1) == BANK(ll)) * 12 + \
        (BANK(\1) == BANK(mm)) * 13 + \
        (BANK(\1) == BANK(nn)) * 14 + \
        (BANK(\1) == BANK(oo)) * 15 + \
        (BANK(\1) == BANK(pp)) * 16 + \
        (BANK(\1) == BANK(zz)) * 26)
ENDM

    test hh
```
```
$ rgbasm file.asm 
zsh: segmentation fault  rgbasm file.asm
```

RGBDS version 0.3.8.

**Comment by AntonioND** [comment]
Can you try making the number in this line bigger? https://github.com/rednex/rgbds/blob/a37a09c09c1f8f5a847ef3117c12b8376d22d916/include/asm/rpn.h#L16 Just to see if it is the problem.

Also, I think that in this case pasting the code here is alright.

**Comment by mid-kid** [comment]
Yeah, increasing that number fixes it.

**Comment by AntonioND** [comment]
Then I guess that the array has to be replaced by a pointer to a malloc'ed buffer that can be resized.

**Comment by mid-kid** [comment]
This was fixed in #335
